libuserhelper.cpp

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (c) 2020 Jolla Ltd.
+ * Copyright (c) 2020 Open Mobile Platform LLC.
+ *
+ * All rights reserved.
+ *
+ * BSD 3-Clause License, see LICENSE.
+ */
+
+#include "libuserhelper.h"
+
+#include <libuser/user.h>
+
+#include <QDebug>
+
+LibUserHelper::LibUserHelper()
+{
+}
+
+uint LibUserHelper::addGroup(const QString &group)
+{
+    struct lu_error *error = nullptr;
+    struct lu_context *context = lu_start(NULL, lu_user, NULL, NULL, lu_prompt_console, NULL, &error);
+
+    if (!context) {
+        qWarning() << "Error creating context:" << lu_strerror(error);
+        lu_error_free(&error);
+        return 0;
+    }
+
+    uint rv = 0;
+    struct lu_ent *ent_group = lu_ent_new();
+    if (lu_group_default(context, group.toUtf8(), false, ent_group)) {
```

Comment:
Is UTF-8 really allowed as a group/user name or does it have to limited to ASCII?

Reply:
New version generates clean user names.